### ISSUE TITLE
Loss function of XGNN

### ISSUE THREAD

**Issue by HyunjuKang** [divelab/DIG#21]
Hello, 

I would like to ask about the loss function of XGNN in gnn_explain.py with a following question:

Specifically at line 100~ 119 in gnn_explain.py
when total_reward < 0 then loss is also negative, but loss should not be below zero.. is there any preprocessing the negative reward? (I hardly find this part of the code)

https://github.com/divelab/DIG/blob/eac506f752fe9394cb33e89d73518bb8382d633a/benchmarks/xgraph/supp/XGNN/gnn_explain.py#L100  

Thank you for your help in advance!

**Comment by Nate1874** [comment]
The loss can be negative. See explanations below. 
The loss function is nn.CrossEntropyLoss(x, y). 
If the reward is positive, minimizing this loss is pushing x towards y. In this case, it means our sampled actions are pretty good and we encourage the generated probability to be closer to the sampled actions. 
If the reward is negative, minimizing the loss is maximizing the CrossEntropyLoss. Then it is pushing pushing x away from y. In this case, the negative reward means the actions are not good, so we need to let the model avoid them.

**Comment by HyunjuKang** [comment]
Thank you!👏

**Comment by HyunjuKang** [comment]
Thank you for the answer.
I learned a lot.

Wish you good luck!

2021년 5월 18일 (화) 오후 1:13, Hao ***@***.***>님이 작성:
